### PR TITLE
(#1265) StartsWith added for Text

### PR DESCRIPTION
**Pull request by ryoku**
This is for #1265

Created class StartsWith
Created tests on StartsWith class

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1268](https://github.com/yegor256/cactoos/issues/1268) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1268](https://github.com/yegor256/cactoos/issues/1268) is assigned to @iakunin[/z](https://www.0crat.com/u/iakunin), here is [why](https://www.0crat.com/footprint/C63314D6Z/ad902092-ef2a-4c8e-a70f-f6ed54667154); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by iakunin** [comment]
@ryoku please, fix all the qulice warnings: https://travis-ci.org/yegor256/cactoos/builds/630436309#L2747-L2783

You can run it locally this way:
```
mvn clean install site -Pqulice -Psite --errors
```

File: src/test/java/org/cactoos/text/StartsWithTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import org.llorllale.cactoos.matchers.ScalarHasValue;
+
+/**
+ * Test case for {@link StartsWith}.
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public class StartsWithTest {
+
+    @Test
+    public void determinesEmptyStartsWithEmpty() {
```
Comment (iakunin) [inline src/test/java/org/cactoos/text/StartsWithTest.java]: @ryoku maybe it's possible to rename test methods without repeating `determines`?

File: src/test/java/org/cactoos/text/StartsWithTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        MatcherAssert.assertThat(
```
Comment (iakunin) [inline src/test/java/org/cactoos/text/StartsWithTest.java]: @ryoku it's better to use EO-matchers (`org.llorllale.cactoos.matchers.Assertion`). See ComparableTextTest for example.

**Comment by iakunin** [comment]
@ryoku now it's much better, thank you. 

One small advice: to increase code coverage it's better to use both ctors in your tests.

**Comment by ryoku** [comment]
@iakunin I've added a test and now coverage looks ok. Thanks for the advice.

**Comment by iakunin** [comment]
@ryoku thank you!

@paulodamaso looks good to merge.

**Comment by fanifieiev** [comment]
@iakunin @paulodamaso Is it fine to locate the `StartsWith` in `org.cactoos.text` rather than in `org.cactoos.scalar` package?

**Comment by iakunin** [comment]
@fanifieiev IMO it's pretty fine, cause `StartsWith` is applicable in the context of `Text` (or `String`).

But let's leave the last word for @paulodamaso

**Comment by fanifieiev** [comment]
> @fanifieiev IMO it's pretty fine, cause `StartsWith` is applicable in the context of `Text` (or `String`).
> 
> But let's leave the last word for @paulodamaso

@iakunin There are many examples of `Scalar ` implementations where they accept `Iterable`, but they are located in the scalar package, not iterable.

**Comment by fanifieiev** [comment]
> > @fanifieiev IMO it's pretty fine, cause `StartsWith` is applicable in the context of `Text` (or `String`).
> > But let's leave the last word for @paulodamaso
> 
> @iakunin There are many examples of `Scalar ` implementations where they accept `Iterable`, but they are located in the scalar package, not iterable.

@iakunin But on the other hand there is also `IsBlank` which resides in `org.cactoos.text`.

**Comment by iakunin** [comment]
> @iakunin But on the other hand there is also `IsBlank` which resides in `org.cactoos.text`.

@fanifieiev exactly :) Let @paulodamaso decide.

**Comment by paulodamaso** [comment]
@fanifieiev @iakunin I think it is ok to keep it in text package, as it uses the string implementation to take the start of it.

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job [#1268](https://github.com/yegor256/cactoos/issues/1268) is not in the agenda of @iakunin[/z](https://www.0crat.com/u/iakunin), can't retrieve data and time of add

**Comment by 0crat** [comment]
Job [#1268](https://github.com/yegor256/cactoos/issues/1268) is not in the agenda of @iakunin[/z](https://www.0crat.com/u/iakunin), can't inspect

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @iakunin[/z](https://www.0crat.com/u/iakunin), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1268](https://github.com/yegor256/cactoos/issues/1268) is now out of scope

**Comment by 0crat** [comment]
Code review was too long (10 days), architects (@paulodamaso) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @iakunin[/z](https://www.0crat.com/u/iakunin)

**Pull request merged**